x-pack/platform/plugins/shared/observability_ai_assistant/server/service/knowledge_base_service/reindex_knowledge_base.ts

```diff
@@ -193,5 +193,5 @@ export async function isReIndexInProgress({
     getActiveReindexingTaskId(esClient),
   ]);
 
-  return lock !== undefined || activeReindexingTask !== undefined;
+  return lock !== undefined && activeReindexingTask !== undefined;
```

Comment:
This was intentional. The lock might have expired because the Kibana node that acquired it crashed. In that case we can still rely on the second check to see if the reindex task exists.

Did you notice any problems with this check since you changed it?